### ISSUE TITLE
VSCodium nap package?

### ISSUE THREAD

**Issue by amankkg** [VSCodium/vscodium#102]
The recent release of VSCode v. 1.33 is now distributed via snap package https://snapcraft.io/code

Would be great to see a snap package for this project also!

**Comment by stripedpajamas** [comment]
I think this is important for us to get working, as the goal would be 100% parity with what is offered by Microsoft. Any help is appreciated toward this goal; I will look into it also when I have some time to.

**Comment by atcommander** [comment]
I would be interested in helping if you are still looking for someone to help with this

**Comment by stripedpajamas** [comment]
Sure!

**Comment by eaglersdeveloper** [comment]
VS Code already has [snapcraft.yaml](https://github.com/Microsoft/vscode/blob/master/resources/linux/snap/snapcraft.yaml). What if we will use it?

**Comment by atcommander** [comment]
We could but it will need to be tweaked

**Comment by mathiasblc** [comment]
> We could but it will need to be tweaked

Can you be more specific about what needs to be done please? I'm willing to help on this.

**Comment by AktasC** [comment]
I'm gonna try to build this for Solus by the end of the month, have a lot of work going on currently.
If you have any suggestions feel free to ping me here.

**Comment by bittner** [comment]
There was a `vscodium` in the "edge" channel for snaps (i.e. automatically built off source). What is the status on this?

Currently, what you get when trying to install vscodium is:
```console
$ snap install vscodium
error: snap "vscodium" is not available on this architecture (amd64) but exists on other
       architectures (i386).
```

**Comment by eaglersdeveloper** [comment]
Who build it?

**Comment by impredicative** [comment]
Sorry but is there any status update for this issue? snap installation will make VSCodium a lot more attractive IMHO.

**Comment by zicklag** [comment]
> Who build it?

It looks like it may have been built from here: https://github.com/ErnyTech/vscodium.snap

**Comment by hackel** [comment]
This needs to become the default (and only) version of VSCode available in the Snap Store.  They do a terrible job of warning users about telemetry/spyware.

**Comment by hivemall** [comment]
@hellsworth is it possible canoncial (beside cherrytree) snapifies vscodium which would be extremly popular (with many privacy concernd users)?

**Comment by hellsworth** [comment]
It is unlikely that Canonical will take up snapping vscodium for two reasons:
1. A user already is providing this in the snap store
2. Canonical has worked with Microsoft to provide vscode in the [snap store](https://snapcraft.io/code)

The best thing to do would be to get the person that published vscodium in the snap store, to transfer the project to be part of [snapcrafters](https://github.com/snapcrafters), which is a community that maintains snaps.

**Comment by bittner** [comment]
@hellsworth With 1. above, do you refer to https://snapcraft.io/vscodium?

Interestingly, this snap is not listed with `snap search`, not sure why. I couldn't see any author information otherwise, so I asked Canonical for support via the "Report this app" link at the bottom of the page.

Now cross your fingers and wait! :crossed_fingers:

**Comment by stripedpajamas** [comment]
@bittner @hellsworth -- I think I'm actually the one who "owns" the vscodium snap. Someone made it initially and then transferred to me. I haven't gotten any time to figure out making the snap hence it has gone stale. 

I'm very uninformed when it comes to snaps. What is the best path forward-- what happens if I were to transfer to snapcrafters?

**Comment by zicklag** [comment]
@bittner I'm pretty sure the reason it is not listed in search is because the snap only has an "edge" release. If we made a stable release it would show in searches.

**Comment by popey** [comment]
@bittner actually it doesn't show because there is only an i386 build in edge. There's no AMD64 build in any channel.

**Comment by popey** [comment]
![Screenshot from 2020-02-11 21-00-34](https://user-images.githubusercontent.com/1841272/74279163-4e804280-4d12-11ea-9635-39dbfbdfa1ff.png)

Ok, just built an up to date VSCodium snap.

Aside: _Is it VSCodium, or Codium? Because there appears to be a split personality going on in the releases area on github :D._ 

Where does this yaml need to live? I could contribute it as a PR to this repo and then someone from the {VS}Codium team could own the snap going forward? If upstream {VS}Codium don't want to maintain the snap, it could go in snapcrafters, but we'd rather upstreams maintained their snaps, as a goal. 

Let me know what you want to do.

**Comment by bittner** [comment]
- The project is called "VSCodium" (just like "VSCode" is called VS Code),
- The binary is meant to be called "codium" (just like "code" is called, w/o the leading "vs").
- The packages (and packagers) on the various platforms do whatever they want. (Hey, it's free software! Let us do what we want! :roll_eyes: :laughing:)

See #28 and #36 for a bit of history.

P.S.: Say hello to Marc Shuttleworth, my hero! :heart_eyes:

**Comment by jobukkit** [comment]
If you search for "vscodium" on Snapcraft without the space, you get an old version.

**Comment by bittner** [comment]
> If you search for "vscodium" on Snapcraft without the space, you get an old version.

This comes from the "edge" channel. With a "normal" query I don't get this when I query snap, e.g.

```console
$ snap search codium
Name    Version  Publisher     Notes    Summary
codium  1.52.1   snapcrafters  classic  Free/Libre Open Source Software Binaries of VSCode
$ snap search vscodium
No matching snaps for "vscodium"
$ snap search "vs codium"
Name    Version  Publisher     Notes    Summary
codium  1.52.1   snapcrafters  classic  Free/Libre Open Source Software Binaries of VSCode
```

The snapcraft.io website shows the [vscodium](https://snapcraft.io/vscodium) snap as from a publisher called "VSCodium". Is this your account, @stripedpajamas?

@popey Can we remove that snap or mark it deprecated?

**Comment by stripedpajamas** [comment]
@bittner yep that's my account, back when I first started investigating how to get a snap together. I've unlisted that old vscodium snap to hopefully reduce the confusion.
